### PR TITLE
A lot improvements

### PR DESCRIPTION
**Pull request by Roka20012**
- Add ability to check country without closing keyboard
- Fix default number formatting
- Set default country US
- Add inputRef
- Now CountryFlag depends of dial code(when phoneNumber is set by default)

### CODE REVIEW

**Comment by Roka20012** [comment]
And can I ask you @MuhammedAlmaz to create `npm published` version of this package?

File: src/IntlPhoneInput.js [hunk 1]
```diff
@@ -119,6 +119,11 @@ export default class IntlPhoneInput extends React.Component {
     this.showModal();
   }
 
+  componentDidMount() {
+    this.props.inputRef.current.focus();
```
Comment (alexstrat) [inline src/IntlPhoneInput.js]: To be removed, it should be userland's role to do this

File: src/IntlPhoneInput.js [hunk 2]
```diff
@@ -18,11 +18,11 @@ import CloseIconAndroid from './close.android.png';
 
 import data from './Countries';
 
-export default class IntlPhoneInput extends React.Component {
+class IntlPhoneInput extends React.Component {
   constructor(props) {
     super(props);
     const defaultCountry =
-        data.filter((obj) => obj.code === props.defaultCountry)[0] || data.filter((obj) => obj.code === "TR")[0];
+         data.filter((obj) =>  props.phoneNumber.includes(obj.dialCode))[0] || data.filter((obj) => obj.code === props.defaultCountry)[0] || data.filter((obj) => obj.code === 'US')[0];
```
Comment (alexstrat) [inline src/IntlPhoneInput.js]: Default country shouldn't be configurable via a prop?

File: src/IntlPhoneInput.js [hunk 3]
```diff
@@ -119,6 +119,11 @@ export default class IntlPhoneInput extends React.Component {
     this.showModal();
   }
 
+  componentDidMount() {
+    this.props.inputRef.current.focus();
+    this.onChangeText(this.state.phoneNumber)
```
Comment (alexstrat) [inline src/IntlPhoneInput.js]: Why? Sure it does not imply weird behavior?

File: src/IntlPhoneInput.js (hunk 2 above)
Comment (Roka20012) [reply src/IntlPhoneInput.js]: Yeap you can set default country via prop, buy if you didn't set any you will get US as default one

File: src/IntlPhoneInput.js (hunk 1 above)
Comment (Roka20012) [reply src/IntlPhoneInput.js]: Done

File: src/IntlPhoneInput.js [hunk 4]
```diff
@@ -136,7 +140,8 @@ export default class IntlPhoneInput extends React.Component {
       filterText,
       searchIconStyle,
       closeButtonStyle,
-      inputProps
+      inputRef,
```
Comment (alexstrat) [inline src/IntlPhoneInput.js:161]: Isn't it necessary to add to prop types?

File: src/IntlPhoneInput.js (hunk 3 above)
Comment (Roka20012) [reply src/IntlPhoneInput.js]: Fixed

File: src/IntlPhoneInput.js (hunk 4 above)
Comment (Roka20012) [reply src/IntlPhoneInput.js:161]: Have added

**Comment by MuhammedAlmaz** [comment]
@Roka20012 i am going to add custom modal on phoneInput component. the default country is not US. TR is fine you can change it with props.

**Pull request closed without merging**